src/coreclr/jit/importer.cpp

```diff
@@ -7993,13 +7993,83 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                 goto COND_JUMP;
 
             case CEE_SWITCH:
+#ifdef DEBUG
+                if (verbose)
+                {
+                    printf("\n---- CEE_SWITCH start in " FMT_BB " (PC=%03u) of '%s'", block->bbNum, block->bbCodeOffs,
+                           info.compFullName);
+                }
+#endif
                 /* Pop the switch value off the stack */
                 op1 = impPopStack().val;
                 assertImp(genActualTypeIsIntOrI(op1->TypeGet()));
 
-                /* We can create a switch node */
+                // Fold Switch for GT_CNS_INT
+                if ((op1->gtOper == GT_CNS_INT) && !compIsForImportOnly())
```

Comment:
```suggestion
                if (opts.OptimizationEnabled() && (op1->gtOper == GT_CNS_INT) && !compIsForImportOnly())
```
We don't want to do this unless we're optimizing the method.

This should fix the diffs you are seeing with minopts.

Reply:
Done.